File: bogo_sort/python/bogoSort.py
# BOGO SORT in Python

import random

def bogoSort(a):
	n = len(a)
	while (is_sorted(a) == False):
		shuffle(a)


def is_sorted(a):
	n = len(a)
	for i in range(0, n-1):
		if (a[i] > a[i+1]):
			return False
	return True


def shuffle(a):
	n = len(a)
	for i in range(0, n):
		r = random.randint(0, n-1)
		a[i], a[r] = a[r], a[i]


# MAIN
a = [5, 4, -3, 2, 1]
bogoSort(a)
print("Sorted array :")
for i in range(len(a)):
	print("%d" % a[i], end=" ")

File: bogo_sort/Java/BogoSort.java
import java.util.*;
 
public class BogoSort{
 
  private static final Random generator = new Random();  
 
  public static void bogoSort(int[] array)  {  
    while (!isSorted(array)) {  
      for (int i = 0; i < array.length; i++){  
        int randomPosition = generator.nextInt(array.length);  
        int temp = array[i];  
        array[i] = array[randomPosition];  
        array[randomPosition] = temp;  
      }  
    }  
  }  
 
  private static boolean isSorted(int[] array)  {  
    for (int i = 1; i < array.length; i++){
      if (array[i] < array[i - 1]) {
        return false;  
      }
    }
    return true;  
  }  
 
  public static void main(String[] args) {
    int [] array = {5,3,0,2,4,1,0,5,2,3,1,4}; 
    System.out.println("Before: " + Arrays.toString(array));
    bogoSort(array);
    System.out.println("After:  " + Arrays.toString(array));
  }
}